multimodal/src/autogluon/multimodal/utils/export.py

```diff
@@ -160,23 +152,85 @@ def export_onnx(
             else:
                 batch[key] = inp.to(device)
         self._model.to(device)
-        input_vec = InputBatch(**batch)
+        keys_to_drop = list(set(batch.keys()).difference(set(self._model.input_keys)))
+        for k in keys_to_drop:
+            batch.pop(k)
+        # Truncate input data types for TensorRT (only support: bool, int32, half, float)
+        if truncate_long_and_double:
+            for k in batch:
+                if batch[k].dtype == torch.int64:
+                    batch[k] = batch[k].to(torch.int32)
+        input_vec = [batch[k] for k in self._model.input_keys]
+
+        self._model.eval()
+
+        if not path:
+            onnx_path = os.path.join(self.path, "model.onnx")
+        else:
+            onnx_path = os.path.join(path, "model.onnx")
+        dirname = os.path.dirname(os.path.abspath(onnx_path))
+        if not os.path.exists(dirname):
+            os.makedirs(dirname)
 
-        with precision_context, torch.no_grad():
-            traced_model = torch.jit.trace(self._model, input_vec)
-        torch.jit.save(traced_model, "traced_model.pt")
-        traced_model = torch.jit.load("traced_model.pt")
+        input_keys = self._model.input_keys
+        dynamic_axes = onnx_get_dynamic_axes(input_keys)
 
         torch.onnx.export(
-            traced_model,
-            args=input_vec,
+            self._model,
+            args=tuple(input_vec),
             f=onnx_path,
             opset_version=opset_version,
             verbose=verbose,
-            input_names=valid_input,
+            input_names=input_keys,
             dynamic_axes=dynamic_axes,
         )
 
+        return onnx_path
+
+    def export_tensorrt(
+        self,
+        data: Optional[pd.DataFrame] = None,
+        path: Optional[str] = None,
+        batch_size: Optional[int] = None,
+    ):
+        """
+        Export this predictor's model to ONNX file.
+
+        Parameters
+        ----------
+        data
+            Raw data used to trace and export the model.
+            If this is None, will check if a processed batch is provided.
+        path
+            The export path of onnx model.
+        batch_size
+            The batch_size of export model's input.
+            Normally the batch_size is a dynamic axis, so we could use a small value for faster export.
+
+        Returns
+        -------
+        trt_module : OnnxModule
+            The onnx-based module that can be used to replace predictor._model for model inference.
+        """
+        import onnx
+        import torch
+
+        from .onnx import OnnxModule
+
+        truncate_long_and_double = False
+        onnx_path = self.export_onnx(
+            data=data, path=path, batch_size=batch_size, truncate_long_and_double=truncate_long_and_double
+        )
+
+        logger.info("Loading ONNX file from path {}...".format(onnx_path))
+        onnx_model = onnx.load(onnx_path)
+
+        trt_module = OnnxModule(onnx_model)
```

Comment:
The returned module may not use tensorrt. Maybe moving the warning of importing tensorrt inside `OnnxModule ` is better?

Reply:
I tried to move importing tensorrt inside OnnxModule, but that didn't work. (onnxruntime.InferenceSession doesn't compile to use TensorrtExecutionProvider)